### DISCUSSION TITLE
Unread notifications

### DISCUSSION THREAD

**Discussion by willin** [community/community#32712] [Other Feature Feedback, Questions, & Ideas] [answered]
![WX20220912-123037@2x](https://user-images.githubusercontent.com/1890238/189574436-53c71d84-c8f9-433c-8993-f5676d6c546f.png)

Inbox shows 1, but not found.

**Comment by radar-blip** [comment]
Its just a glitch, nothing to worry about. To get rid of it, mark all your notifications as Read.

**Reply by willin** [reply]
no use...

**Reply by radar-blip** [reply]
They all have to be marked 'Read' even the ones that are marked as 'Done'.

**Comment by airtower-luna** [comment, marked as answer]
I recently had that issue, too, it's a known bug. You can mark the ghost notification as read using the [notifications API](https://docs.github.com/en/rest/activity/notifications#mark-notifications-as-read), you need a token with *notifications* scope to authorize the request.

I used the Python script below, which lists unread notifications before marking them as read and generally has some more output because I'm curious. It expects the token in the `GH_TOKEN` environment variable. :smile_cat: 

```py
import json
import os
import requests
from datetime import datetime, timezone

NOTIFICATIONS = 'https://api.github.com/notifications'

token = os.environ['GH_TOKEN']
s = requests.Session()
s.headers.update({
    'Authorization': f'Bearer {token}',
    'Accept': 'application/vnd.github+json'})

r = s.get(NOTIFICATIONS)
print(json.dumps(r.json(), indent=2))

now = datetime.now(timezone.utc)
print(now.isoformat(timespec='seconds'))

r = s.put(
    NOTIFICATIONS,
    json={
        'last_read_at': now.isoformat(timespec='seconds'),
        'read': True
    })
print(r.status_code)
for k, v in r.headers.items():
    print(f'{k}: {v}')
print(r.text)
```

**Reply by jleverenz** [reply]
Thanks for sharing the script! For me, looks like it cleared out the "stuck" notification from the Unread list, but it still shows up as "not done" in some parts of the UI. A smaller issue, but fyi for others. e.g. https://github.com/notifications?query=-is%3Adone still shows `1-0 of 1` with nothing in the list. 

And marking a notification done isn't supported by the API yet, it seems: https://github.com/community/community/discussions/15669

**Reply by airtower-luna** [reply]
That was the same for me, and a known issue according to the person who responded to my bug report. :sweat_smile: For me the notification reappeared (as read) after a couple hours and I could mark it as done then, not sure how common that is.

**Reply by javasgl** [reply]
The world is finally clear！

**Reply by KubaJastrz** [reply]
Thanks, it worked

**Comment by willin** [comment]
<img width="860" alt="image" src="https://user-images.githubusercontent.com/1890238/189786269-1f55885d-1166-4bac-a05c-2a974e5e643f.png">

**Comment by Algomorph** [comment]
Sorry for extra notifications in thread, but this is just such a funny bug.
![image](https://github.com/community/community/assets/1050606/5cc1fbbe-70cc-4452-aa07-3a0d01d8d808)

**Comment by skyhirider** [comment]
Have this bug now, so seems it is still not fixed.

On Windows, this worked (after downloading curl and not that broken facade powershell is using by default).

Also notice the double quotes in the json being escaped by a backslash, terminal will send out the wrong data otherwise.

```
curl.exe -L -X PUT -H "Accept: application/vnd.github+json" -H "Authorization: Bearer putYourTokenHere" -H "X-GitHub-Api-Version: 2022-11-28" https://api.github.com/notifications -d '{\"last_read_at\":\"2025-03-10T00:00:00Z\",\"read\":true}'
```

Source - https://docs.github.com/en/rest/activity/notifications?apiVersion=2022-11-28#mark-notifications-as-read
